File: Assignment51/program1.cpp
/*
1.  Write a generic program which accept one value and one number from user.
    Print that value that number of times on screen.
*/

#include<iostream>
using namespace std;

template<class T>

void Display(T value, int iSize)
{
    int iCnt = 0;
    for(iCnt = 1; iCnt<=iSize; iCnt++)
    {
        cout<<value<<"\t";
    }
    cout<<"\n";
}

int main()
{
    Display('M', 7);
    Display(11, 3);
    Display(3.7, 6);
    

    return 0;
}

File: Assignment58/program1.java
//1. Write java program to accept directory name from user and display all names of files from that directory which are regular file.
import java.io.File;
import java.util.Scanner;

class program1
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter directory name: ");
        String dirName = sobj.nextLine();

        File dir = new File(dirName);

        if (!dir.exists() || !dir.isDirectory())
        {
            System.out.println("Invalid directory");
            sobj.close();
            return;
        }

        File files[] = dir.listFiles();

        if (files == null || files.length == 0)
        {
            System.out.println("Directory is empty");
        }
        else
        {
            System.out.print("Files in directory " + dirName + ":\n");

            for (int i = 0; i < files.length; i++)
            {
                File f = files[i];
                if (f.isFile())
                    System.out.println(f.getName() + " (" + f.length() + " bytes)");
            }
        }

        sobj.close();
    }
}
